lib/migrations/migrate/Migrator.js

```diff
@@ -222,7 +228,13 @@ class Migrator {
         });
 
         let migrationToRun;
-        const name = this.config.name;
+        const name =
+          this.config.name != null
+            ? this.config.migrationSource.getMigrationName({
+                file: this.config.name,
+                directory: this.config.directory,
+              })
+            : null;
```

Comment:
duplicate code with above

Reply:
Created a function to contain the duplicate code